wikimedia/wikipedia-ios
# Remove Carthage and resolve current dependency management issues

**Phabricator:** https://phabricator.wikimedia.org/T275557

### Notes

This PR fixes our current dependency management issues and removes Carthage. It moves the dependencies that were previously in Carthage directly into the source of the project. Additionally, this move from Universal Frameworks from Carthage to integrating the source in the project comes with the bonus that this project now compiles on both Intel and Apple Silicon Macs.

The [CodeMirror repo](https://github.com/wikimedia/wikipedia-ios-codemirror) was also previously included in the Cartfile, but has now been moved to a Git submodule in this project. To clone the repo with the submodules use `git clone --recursive https://github.com/wikimedia/wikipedia-ios.git`, or you can use the new `scripts/update_submodules` script to initialize the submodule on already cloned copies.

Apologies for the large PR. I strongly recommend reviewing it by going commit by commit – I've tried to go step by step with smaller easy to follow commits to provide a coherent picture of what changed.

### Test Steps

Because this is high touch in terms of dependencies on existing targets, it's recommended to do a shallow regression test of app features and the extensions.

1. Confirm `scripts/setup` works
2. Confirm Wikipedia and Experimental targets build and deploy to Simulator and Device
3. Confirm main Wikipedia target runs unit tests (`testSmallViewModelCorrectlyInstantiates` currently failing is expected behavior)
4. Do shallow regression test of Wikipedia target deployment's features (app features, extensions like old and new widgets)
5. Confirm viewing images in gallery works on Wikipedia target
6. Confirm Wikipedia target archives for device and validates with App Store
